wallets/react-wallet-v2/src/utils/PolkadotRequestHandlerUtil.ts

```diff
@@ -10,15 +10,20 @@ export async function approvePolkadotRequest(
 ) {
   const { params, id } = requestEvent
   const { request } = params
-  const wallet = polkadotWallets[getWalletAddressFromParams(polkadotAddresses, params)]
+  const address = request.params?.address
+  const wallet = getPolkadotWallets(address)
+
+  if (!wallet) {
+    throw new Error('Wallet does not exist')
+  }
 
   switch (request.method) {
     case POLKADOT_SIGNING_METHODS.POLKADOT_SIGN_MESSAGE:
-      const signature = await wallet.signMessage(request.params.message)
+      const signature = await wallet?.signMessage(request.params.message)
```

Comment:
Is the optional chaining needed with the reference guard (`if (!wallet)`) above?

Reply:
right. can be removed.